operator/cmd/mesh/profile-common.go

```diff
@@ -159,8 +178,48 @@ func genIOPS(inFilename []string, profile, setOverlayYAML, ver string, force boo
 	return finalYAML, finalIOPS, nil
 }
 
-func genProfile(helmValues bool, inFilename []string, profile, setOverlayYAML, configPath string, force bool, l *Logger) (string, error) {
-	finalYAML, finalIOPS, err := genIOPS(inFilename, profile, setOverlayYAML, "", force, l)
+func getClusterSpecificValues(config *rest.Config, force bool, l *Logger) (string, error) {
+	overlays := []string{}
+
+	jwt, err := getJwtTypeOverlay(config, l)
+	if err != nil {
+		if force {
+			l.logAndPrint(err)
+		} else {
+			return "", err
+		}
+	} else {
+		overlays = append(overlays, jwt)
+	}
+
+	return MakeTreeFromSetList(overlays, false, l)
+
+}
+
+func getJwtTypeOverlay(config *rest.Config, l *Logger) (string, error) {
+	d, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return "", fmt.Errorf("failed to determine JWT policy support. Use the --force flag to ignore this: %v", err)
+	}
+	_, s, err := d.ServerGroupsAndResources()
+	if err != nil {
+		return "", fmt.Errorf("failed to determine JWT policy support. Use the --force flag to ignore this: %v", err)
+	}
+	for _, res := range s {
+		for _, api := range res.APIResources {
+			// Appearance of this API indicates we do support third party jwt token
+			if api.Name == "serviceaccounts/token" {
+				return "values.global.jwtPolicy=third-party-jwt", nil
+			}
+		}
+	}
+	// TODO link to istio.io doc on how to secure this
+	l.logAndPrint("Detected that your cluster does not support third party JWT authentication. Falling back to less secure first party JWT")
```

Comment:
i recommend separating the printing from the actual logic - this will make it easier to use the same code in the controller. ideally the cluster querying logic would live in pkg (kubectlcmd is one possible place).